icu4c/source/common/brkiter.cpp

```diff
@@ -535,9 +547,11 @@ int32_t BreakIterator::getRuleStatusVec(int32_t *fillInVec, int32_t capacity, UE
     return 1;
 }
 
-BreakIterator::BreakIterator (const Locale& valid, const Locale& actual) {
+BreakIterator::BreakIterator (const Locale& valid, const Locale& actual) : actualLocale(new CharString()), validLocale(new CharString), requestLocale(new CharString()) {
   U_LOCALE_BASED(locBased, (*this));
-  locBased.setLocaleIDs(valid, actual);
+  UErrorCode status = U_ZERO_ERROR;
+  locBased.setLocaleIDs(valid, actual, status);
```

Comment:
Before this line, please check if any of the three pointers is a nullptr. If so, then set the memory-allocation-error.